pkg/generators/system/sphinx_service.go

```diff
@@ -21,7 +21,7 @@ func (gen *SphinxGenerator) Service() basereconciler.GeneratorFunction {
 				APIVersion: corev1.SchemeGroupVersion.String(),
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      gen.GetComponent(),
+				Name:      gen.DatabaseService,
```

Comment:
Why changing the name of the Service resource? This would trigger delete + create of the service whenever a change occurs to the field `spec.config.sphinxAddress`

Reply:
To set the service name both in the sidekiq/app and the k8s sphinx service.

Reply:
it means that, if you somehow use an external sphinx, and configure `sphinxAddress=sphinx.va.3sca.net` with that external sphinx, the operator will try to create a service whose name would be sphinxAddress`value?

Reply:
`spec.config.sphinxAddress` just affects the Service name and has no other funtionality attached, it should be called `spec.config.sphinxServiceName` for clarity.

Reply:
So my question is:
- If we make customizable via CRD field the possible sphinx service name, it is because it can be used an external sphinx (and then, don't deploy the internal sphinx)
- But after looking into the code changes, iI understand that the service name to put into the new field, will be used to configure the k8s service that the operator will create (so internal, nothing external), so what's the point of being able to customizing it instead of harcoding it?
- It is that correct or maybe I'm missing something?

Reply:
Regarding the naming, it affects to the `THINKING_SPHINX_ADDRESS` env var for `system-app` and `system-sidekiq` and consequently, to the `serviceName` in case the `system-sphinx` is deployed by the operator (that's always the case in the scope of this fix). I renamed about 10 times the var (sphinxAddress, sphinxService, sphinxEndpoint,...) and opted for `sphinxAddress` to match the **env var** like the rest of the variables. The `address`/`port` pair is a quite common naming convention for database endpoints. It's used also as service name because is "propagated" from the `system-app` and `system-sidekiq` options configuration. Anyhow, I don't have any particular attachment to that name, we can change it.

If the need of using an external service arises in the future, the sphinx sts deployment should be optional, but `system-app` and `system-sidekiq` will require still require a `sphinxAddress` and a `sphinxPort`. But that functionality is not currently required and it's outside the scope of this fix. Is customisable but defaulted to `system-sphinx`.

Reply:
I really think it should be `sphinxServiceName`. Naming it `sphinxAddress` gives the impression that you can deploy sphinx on your own and just point to it with this variable.

Reply:
As discussed internally, the parametrisation options for that value have been removed and now is just a constant: https://github.com/3scale-ops/saas-operator/pull/106/commits/a18ae58c095cf544c215c438071db4a304b02956